sycl/include/sycl/ext/oneapi/experimental/graph.hpp

```diff
@@ -535,6 +547,42 @@ class dynamic_parameter : public detail::dynamic_parameter_base {
   }
 };
 
+template <typename DataT, int Dimensions = 1>
+class dynamic_local_accessor : public detail::dynamic_parameter_base {
```

Comment:
We chatted about inheriting from a separate `dynamic_local_accessor_base` class, rather than having the dynamic parameter base & impl classes doing all the different types of dynamic parameter objects, and didn't really come to any conclusion. 

Out of interest, did you try prototyping that at all? Not that I'm sure if I have a preference for it